### PR TITLE
Resolve rubocop offenses for auth_controller.rb

### PR DESCRIPTION
**Pull request by Eschon**
Since the last one was merged so fast I decided to do another file.
This time I'm not sure about how I fixed the long lines. Does anyone have a better idea?

### CODE REVIEW

File: app/controllers/auth_controller.rb [hunk 1]
```diff
@@ -8,7 +8,10 @@ def sign_out_action
   end
 
   def sign_in_action
-    return error! "Missing parameters", 400 unless [:email, :password].none? { |x| params[x].blank? }
+    return error! 'Missing parameters', 400 unless %i(email password)
```
Comment (NuckChorris) [inline app/controllers/auth_controller.rb]: Probably best to split it to a multi-line conditional

File: app/controllers/auth_controller.rb [hunk 2]
```diff
@@ -17,20 +20,25 @@ def sign_in_action
...
 
   def sign_up_action
-    return error! "Missing parameters", 400 unless [:username, :email, :password].none? { |x| params[x].blank? }
-    return error! "User with that email already exists", 409 if User.exists?(['lower(email) = ?', params[:email].downcase])
-    return error! "User with that name already exists", 409 if User.exists?(['lower(name) = ?', params[:username].downcase])
+    return error! 'Missing parameters', 400 unless %i(username email password)
+                                                   .none? do |x|
+                                                     params[x].blank?
+                                                   end
+    return error! 'User with that email already exists', 409 if User.exists?(
```
Comment (NuckChorris) [inline app/controllers/auth_controller.rb]: These should probably get offloaded to model methods, but if you don't wanna bother right now this works well enough

File: app/controllers/auth_controller.rb (hunk 1 above)
Comment (Eschon) [inline app/controllers/auth_controller.rb]: How would I do that? Won't rubocop complain when I change it to
```Ruby
unless [:email, :password].none? { |x| params[x].blank? }
  return error! 'Missing parameters', 400
end
```

Comment (NuckChorris) [inline app/controllers/auth_controller.rb]: I _hope_ it's smart enough to tell that it would violate another rule to flip it… but I don't know

**Comment by Eschon** [comment]
I changed the long conditionals like @NuckChorris suggested and rubocop was ok with it.
I squashed everything so feel free to merge if nothing else needs to be changed.

**Comment by NuckChorris** [comment]
:+1: You're really tearing through these! LGTM!

**Comment by Eschon** [comment]
I don't have that much free time but sometimes I still don't know what to do with it :sweat_smile:

**Pull request merged**